src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1492,24 +1493,30 @@ private boolean isConsumerPersonNameValid(String name) {
      */
     // TODO: Re-evaluate if this is still an issue with the new membership
     // scheme!
-    private void createOwnerIfNeeded(Principal principal) {
-        if (!(principal instanceof UserPrincipal)) {
-            // If this isn't a user principal we can't check for owners that may need to be created.
-            return;
-        }
-
-        for (Owner owner : ((UserPrincipal) principal).getOwners()) {
-            Owner existingOwner = ownerCurator.getByKey(owner.getKey());
-
-            if (existingOwner == null) {
-                log.info("Principal carries permission for owner that does not exist.");
-                log.info("Creating new owner: {}", owner.getKey());
-
-                existingOwner = ownerCurator.create(owner);
-
-                this.refresherFactory.getRefresher(this.subAdapter, this.prodAdapter)
-                    .add(existingOwner)
-                    .run();
+    private void createOwnerIfNeeded(Principal principal, String ownerKey) {
+        if (principal instanceof TrustedUserPrincipal &&
+            !this.ownerCurator.existsByKey(ownerKey)) {
+            Owner ownerToCreate = new Owner()
+                .setKey(ownerKey)
+                .setDisplayName(ownerKey)
+                .setContentAccessModeList(ContentAccessManager.getListDefaultDatabaseValue())
+                .setContentAccessMode(ContentAccessMode.ORG_ENVIRONMENT.toDatabaseValue());
+            this.ownerCurator.create(ownerToCreate);
```

Comment:
Actually, thinking about this a big more, let's not refresh the owner directly like the UserPrincipal. Instead run a `RefreshPoolsJob`. This will prevent any conflicts if such a job is already running at the same time, at the job constraint level

Reply:
Done.